tests/test-location-label.c

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2020 Endless Mobile, Inc. */
+
+/* This file is part of eos-metrics-instrumentation.
+ *
+ * eos-metrics-instrumentation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-metrics-instrumentation is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-metrics-instrumentation.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include "eins-location-label.h"
+
+static void
+test_empty_keyfile (void)
+{
+  g_autoptr(GKeyFile) kf = g_key_file_new ();
+  g_autoptr(GVariant) payload = build_location_label_event (kf);
```

Comment:
It's a floating GVariant so it doesn't make sense to use `g_autoptr` right?

Reply:
(and similarly for the other tests)

Reply:
It's floating, but, it still has a ref count of 1, and unreffing it will still release it. (As far as I understand from the documentation and the source.)

Reply:
I adjusted the tests to take the floating reference where appropriate, and drop `g_autoptr` in the "it's NULL" cases.